### PR TITLE
[bitnami/etcd] Various small improvements

### PR DESCRIPTION
**Pull request by pantaoran**
**Description of the change**

- _helpers.tpl: I don't see why the path of the dataDir should be different depending on whether persistence is enabled or not. This simplifies things.
- scripts-configmap.yaml: It was my experience that the volumeMount one level above the dataDir erased many files during the mounting of the empty persistence directory. My changes make it so that only the actual dataDir directly is mounted to prevent erasing files at the parent level. I also fixed some problems with unescaped double quotes.
- statefulset.yaml: Fixed some hardcoded paths to use the existing variable. Added an empty /tmp directory volume for environments like mine where containers are fully read-only by policy and only mounted volumes are exempted from that. Adding this volume will not change anything for writeable environments, but it's needed for read-only environments.
- values-*.yaml: I added the possibility to have additional securityContext options taken into consideration by using the entire dictionary in the template. In my environment for example the `runAsNonRoot: true` key is mandatory. For this I had to move the `enable` key out of the dictionary into its own key.

**Benefits**

See description above.

**Possible drawbacks**

None that I know of.

**Applicable issues**

I didn't open issues for these first.

**Checklist**
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

File: bitnami/etcd/values.yaml [hunk 1]
```diff
@@ -129,10 +129,13 @@ auth:
 ## Kubernetes Security Context
 ## https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
 ##
+securityContextEnabled: true
 securityContext:
-  enabled: true
```
Comment (juan131) [inline bitnami/etcd/values.yaml]: Please revert this change. It includes `backwards` incompatibilities.

File: bitnami/etcd/templates/statefulset.yaml [hunk 2]
```diff
@@ -280,6 +281,8 @@ spec:
         configMap:
           name: {{ template "etcd.fullname" . }}-scripts
           defaultMode: 0755
+      - name: tmp
+        emptyDir: {}
```
Comment (juan131) [inline bitnami/etcd/templates/statefulset.yaml]: Why do we need this volume?

File: bitnami/etcd/templates/_helpers.tpl [hunk 3]
```diff
@@ -79,12 +79,8 @@ Return the proper etcd client protocol
 Return the proper etcd data dir
 */}}
 {{- define "etcd.dataDir" -}}
-{{- if .Values.persistence.enabled -}}
-{{- print "/bitnami/etcd/data" -}}
-{{- else -}}
 {{- print "/opt/bitnami/etcd/data" -}}
 {{- end -}}
-{{- end -}}
```
Comment (juan131) [inline bitnami/etcd/templates/_helpers.tpl]: Did you test this change? What happens when you disable the persistence?

File: bitnami/etcd/templates/statefulset.yaml (hunk 2 above)
Comment (juan131) [reply bitnami/etcd/templates/statefulset.yaml]: I just saw the explanation in your summary. Ignore this comment please

File: bitnami/etcd/values.yaml (hunk 1 above)
Comment (pantaoran) [reply bitnami/etcd/values.yaml]: Sorry could you elaborate? How is it incompatible?

File: bitnami/etcd/templates/_helpers.tpl (hunk 3 above)
Comment (pantaoran) [reply bitnami/etcd/templates/_helpers.tpl]: Worked fine when I quickly deployed without persistence for a smoke test.

File: bitnami/etcd/values.yaml (hunk 1 above)
Comment (juan131) [reply bitnami/etcd/values.yaml]: It modifies the structure of the **values.yaml**. 

It's true it doesn't imply a major change but a user who's using a specific **values.yaml** needs to adapt to continue enabling the securityContext. Therefore, if we want to go ahead with the change (at least) we need to mention the change in the README.md (Upgrading section) to alert users about the change.

**Comment by andresbono** [comment]
Hi @pantaoran, what do you think about @juan131's comment here https://github.com/bitnami/charts/pull/1332#discussion_r312022136?

**Comment by andresbono** [comment]
Hi again @pantaoran, just pinging you again.

If we don't get an answer, we may apply some minor changes to the PR and merge it so that your work is not lost!

**Comment by raoofm** [comment]
@pantaoran @juan131 @andresbono just checking if this was supposed to be closed or just the bot accidentally did it.

**Comment by carrodher** [comment]
It was not accidentally, there are some incomplete requests, so we can't merge this PR until the suggestions are applied. To avoid the accumulation of old Issues/PRs, the bot closes it after some days without answers.
